packages/application-generic/src/factories/mail/mail.factory.ts

```diff
@@ -48,7 +49,12 @@ export class MailFactory {
   ): IMailHandler {
     const handler =
       this.handlers.find((handlerItem) =>
-        handlerItem.canHandle(integration.providerId, integration.channel)
+        handlerItem.canHandle(
```

Comment:
I think this is the base place to handle it now days. It was not here before because then we used the stateless library instead and that one was much more used by others.

Reply:
Have you considered creating a NovuEmailHandler to avoid the condition and where we can store the provider used under the hood?
Also with that we might be able to clean up a bit how we avoid the self hosted users to not have it.

Reply:
No but I like your suggestion so will do that :)